src/libraries/System.Runtime/tests/System/Reflection/AssemblyTests.cs

```diff
@@ -0,0 +1,1563 @@
+            yield return new object[] { typeof(int).Module, typeof(int).Assembly };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetAssembly_Custom_TestData))]
+        public void GetAssembly_InvokeCustom_ReturnsExpected(Module module, Assembly expected)
+        {
+            var type = new CustomType
+            {
+                ModuleResult = module
+            };
+            Assert.Same(expected, Assembly.GetAssembly(type));
+        }
+
+        [Fact]
+        public void GetAssembly_NullType_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>("type", () => Assembly.GetAssembly(null));
+        }
+
+        [Fact]
+        public void LoadFrom_NullAssemblyFile_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>("assemblyFile", () => Assembly.LoadFrom(null));
+        }
+
+        [Theory]
+        [InlineData("")]
+        [InlineData("\0")]
+        public void LoadFrom_InvalidAssemblyFile_ThrowsArgumentException(string assemblyFile)
+        {
+            Assert.Throws<ArgumentException>("path", () => Assembly.LoadFrom(assemblyFile));
+        }
+
+        [Fact]
+        public void LoadFrom_NoSuchAssemblyFile_ThrowsFileNotFoundException()
+        {
+            Assert.Throws<FileNotFoundException>(() => Assembly.LoadFrom("NoSuchAssembly"));
+        }
+
+        public static IEnumerable<object[]> LoadFrom_String_ByteArray_AssemblyHashAlgorithm_TestData()
+        {
+            yield return new object[] { null, null, System.Configuration.Assemblies.AssemblyHashAlgorithm.None - 1 };
+            yield return new object[] { string.Empty, Array.Empty<byte>(), System.Configuration.Assemblies.AssemblyHashAlgorithm.None };
+            yield return new object[] { "assemblyFile", new byte[] { 1, 2, 3 }, System.Configuration.Assemblies.AssemblyHashAlgorithm.MD5 };
+        }
+
+        [Theory]
+        [MemberData(nameof(LoadFrom_String_ByteArray_AssemblyHashAlgorithm_TestData))]
+        public void LoadFrom_InvokeStringByteArrayAssemblyHashAlgorithm_ThrowsNotSupportedException(string assemblyFile, byte[] hashValue, System.Configuration.Assemblies.AssemblyHashAlgorithm hashAlgorithm)
+        {
+            Assert.Throws<NotSupportedException>(() => Assembly.LoadFrom(assemblyFile, hashValue, hashAlgorithm));
+        }
+
+        public static IEnumerable<object[]> ReflectionOnlyLoad_ByteArray_TestData()
+        {
+            yield return new object[] { null };
+            yield return new object[] { Array.Empty<byte>() };
+            yield return new object[] { new byte[] { 1, 2, 3 } };
+        }
+
+        [Theory]
+        [MemberData(nameof(ReflectionOnlyLoad_ByteArray_TestData))]
+        public void ReflectionOnlyLoad_InvokeByteArray_ThrowsPlatformNotSupportedException(byte[] rawAssembly)
+        {
+            Assert.Throws<PlatformNotSupportedException>(() => Assembly.ReflectionOnlyLoad(rawAssembly));
```

Comment:
I think its good to have these tests for parameters because in some cases (e.g. `System.Runtime.InteropServices.Marshal`) we actually perform argument validation then throw PNSE

Reply:
Ok